src/components/Nav/Navigation.tsx

```diff
@@ -98,42 +103,68 @@ class Navigation extends React.Component<PropsType, StateType> {
     }
   };
 
+  authenticated = (logged: boolean) => {
+    this.setState({ authenticated: logged });
+    let className = 'login-pf';
+    if (logged) {
+      className = 'layout-pf layout-pf-fixed';
+    }
+    document.documentElement.className = className;
+  };
+
   render() {
+    store.subscribe(() => {
+      const actualState = store.getState() || {};
+      if (actualState['authentication']['user'] !== undefined) {
+        this.authenticated(true);
+      }
+    });
+
     return (
       <>
-        <VerticalNav
-          setControlledState={this.setControlledState}
-          activePath={this.state.selectedItem}
-          navCollapsed={this.state.navCollapsed}
-        >
-          <VerticalNav.Masthead title="Kiali">
-            <VerticalNav.Brand iconImg={pfLogo} titleImg={pfBrand} />
-            <VerticalNav.IconBar>
-              <MessageCenter.Trigger />
-              <HelpDropdown />
-            </VerticalNav.IconBar>
-            <MessageCenter drawerTitle="Message Center" />
-          </VerticalNav.Masthead>
-          <VerticalNav.Item title={serviceGraphTitle} iconClass="fa pficon-topology" onClick={this.navigateTo} />
-          <VerticalNav.Item title={servicesTitle} iconClass="fa pficon-service" onClick={this.navigateTo} />
-          <VerticalNav.Item title={istioConfigTitle} iconClass="fa pficon-template" onClick={this.navigateTo} />
-          <VerticalNav.Item title={servicesJaeger} iconClass="fa fa-paw" onClick={this.navigateTo} />
-        </VerticalNav>
-        <SwitchErrorBoundary
-          fallBackComponent={() => (
-            <PfContainerNavVertical>
-              <h2>Sorry, there was a problem. Try a refresh or navigate to a different page.</h2>
-            </PfContainerNavVertical>
-          )}
-        >
-          <Route path="/service-graph/:namespace" component={ServiceGraphRouteHandler} />
-          <Route path={servicesPath} component={ServiceListPage} />
-          <Route path={servicesJaegerPath} component={ServiceJaegerPage} />
-          <Route path="/namespaces/:namespace/services/:service" component={ServiceDetailsPage} />
-          <Route path={istioConfigPath} component={IstioConfigPage} />
-          <Route path="/namespaces/:namespace/istio/:objectType/:object" component={IstioConfigDetailsPage} />
-          <Redirect to={serviceGraphPath} />
-        </SwitchErrorBoundary>
+        {this.state.authenticated ? (
```

Comment:
I think you can remove some unnecessary div here to reduce the DOM depth, with something like:
```
    return this.state.authenticated ? (
      <>
        <VerticalNav
           blablabla
      </>
    ) : <ConnectedLoginPage dispatch={undefined} />;
```